src/main/MQ2ImGuiConsole.cpp

```diff
@@ -902,6 +904,142 @@ namespace imgui
+			if (sqlite3_exec(db, "CREATE TABLE IF NOT EXISTS entries (entry_timestamp TEXT, pid INTEGER, command TEXT)", nullptr, nullptr, &err_msg) != SQLITE_OK)
+			{
+				WriteChatf("MQ Console Error creating console buffer table: %s", err_msg);
+				sqlite3_free(err_msg);
+				sqlite3_close(db);
+				db = nullptr;
+			}
+			else
+			{
+				if (sqlite3_exec(db, "PRAGMA journal_mode=WAL;", nullptr, nullptr, &err_msg) != SQLITE_OK)
+				{
+					WriteChatf("MQ Console Error setting console buffer journal mode: %s", err_msg);
+					sqlite3_free(err_msg);
+					sqlite3_close(db);
+					db = nullptr;
+				}
+				else
+				{
+					// Fill the history buffer prioritizing this PID and limiting result sets from other PIDs.
+					const char* query = R"(
+						WITH PriorityEntries AS (
+							SELECT entry_timestamp,
+								pid,
+								command,
+								1 AS Priority,
+								MAX(entry_timestamp) OVER() AS LastTimestamp
+							FROM entries
+							WHERE pid = ? -- Prioritize the current PID
+						),
+
+						OtherPIDs AS (
+							SELECT pid,
+								MAX(entry_timestamp) AS LastTimestamp
+							FROM entries
+							WHERE pid != ? -- exclude the current PID
+							GROUP BY pid
+							ORDER BY LastTimeStamp DESC
+							LIMIT 3 -- only get the last 3 processes
+						),
+
+						OtherEntries AS (
+							SELECT oe.entry_timestamp,
+								oe.pid,
+								oe.command,
+								2 AS Priority,
+								op.LastTimeStamp
+							FROM entries oe
+							INNER JOIN OtherPIDs op ON oe.pid = op.pid
+						)
+
+						SELECT command
+						FROM (
+							SELECT *
+							FROM PriorityEntries
+
+							UNION ALL
+
+							SELECT *
+							FROM OtherEntries
+							ORDER BY LastTimestamp DESC -- get the latest entries first
+							LIMIT 50 -- we only want 50 commands since this isn't our original pid anyway
+						)
+						--ORDER BY Priority ASC, LastTimeStamp DESC, pid ASC, entry_timestamp DESC
+						ORDER BY
+							Priority DESC,       -- Sort by the current PID first
+							LastTimeStamp ASC,   -- Then sort by the last timestamp of the other PIDs
+							pid DESC,            -- Then sort by the PID in case there's a tie
+							entry_timestamp ASC  -- Then sort by the timestamp of the entry so they're in order
+					)";
+
+					sqlite3_stmt* stmt;
+			        if (sqlite3_prepare_v2(db, query, -1, &stmt, nullptr) != SQLITE_OK)
+			        {
+						WriteChatf("MQ Console Error preparing query for console buffer retrieval: %s", sqlite3_errmsg(db));
+					}
+					else
+					{
+						sqlite3_bind_int(stmt, 1, process_id);
+						sqlite3_bind_int(stmt, 2, process_id);
+
+						while (sqlite3_step(stmt) == SQLITE_ROW)
+						{
+							if (const char* text = reinterpret_cast<const char*>(sqlite3_column_text(stmt, 0)))
+							{
+								history.emplace_back(text);
+							}
+						}
+
+						sqlite3_finalize(stmt);
+					}
+				}
+			}
+		}
+	}
+	return history;
+}
+
+void AddEntryToDatabase(sqlite3*& db, int process_id, const char* entry)
```

Comment:
this doesn't appear to write to db, so should'nt be reference?

Reply:
Good catch.  Had it as a reference when I added the menu because I was going to have AddEntryToDatabase init the DB if the checkbox was checked but I didn't like that implementation and backed it out.